sympy/stats/stochastic_process_types.py

```diff
@@ -274,7 +287,7 @@ def __call__(self, time):
         if not time.is_symbol and time not in self.index_set:
             raise IndexError("%s is not in the index set of %s"%(time, self.symbol))
         func_obj = Function(self.symbol)(time)
-        pspace_obj = StochasticPSpace(self.symbol, self, self.distribution)
+        pspace_obj = StochasticPSpace(self.symbol, self, distribution=self.distribution(key=time))
```

Comment:
Why are we explicitly passing `distribution=self.distribution(key=time)`? Simply `self.distribution(time)` also works

Reply:
> Simply `self.distribution(time)` also works

Yes.